crates/next_version/src/version_updater.rs

```diff
@@ -128,6 +133,76 @@ impl VersionUpdater {
         self
     }
 
+    /// Configures a custom regex pattern for major version increments.
+    /// this will check only the type of the commit against the given pattern.
+    ///
+    /// ### Note
+    /// `commit type` according to the spec is only `[a-zA-Z]+`
+    ///
+    /// Default: `None`.
+    ///
+    /// ```rust
+    /// use semver::Version;
+    /// use next_version::VersionUpdater;
+    ///
+    /// let commits = ["abc: incompatible change"];
+    /// let version = Version::new(1, 2, 3);
+    /// assert_eq!(
+    ///     VersionUpdater::new()
+    ///         .with_custom_major_increment_regex("abc")
+    ///         .increment(&version, &commits),
+    ///     Version::new(2, 0, 0)
+    /// );
+    /// assert_eq!(
+    ///     VersionUpdater::new()
+    ///         .increment(&version, &commits),
+    ///     Version::new(1, 2, 4)
+    /// );
+    /// ```
+    pub fn with_custom_major_increment_regex(mut self, custom_major_increment_regex: &str) -> Self {
+        if let Ok(regex) = Regex::new(custom_major_increment_regex) {
+            self.custom_major_increment_regex = Some(regex);
+        } else {
+            eprintln!("Invalid regex pattern provided for custom major increment");
+        }
+        self
+    }
```

Comment:
for these `with_custom_` functions, let's return the regex error to the caller, so that they can check if the operation was ok or not 👍

Reply:
done 5f8eef9